transaction-volume-anomaly-detection/src/agent.js

```diff
@@ -0,0 +1,375 @@
+const {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  getEthersProvider,
+  getTransactionReceipt,
+} = require("forta-agent");
+
+const {
+  getContractsByChainId,
+  globalSensitivity,
+  getBlocktimeByChainId,
+  getMinBucketBlockSizeByChainId,
+} = require("./agent.config");
+
+let { bucketBlockSize } = require("./agent.config");
+
+const ARIMA_SETTINGS = {
+  p: 2,
+  d: 1,
+  q: 2,
+  P: 1,
+  D: 0,
+  Q: 1,
+  s: 50,
+  verbose: false,
+};
+let blockTime = 0;
+const ARIMA = require("arima");
+const provider = getEthersProvider();
+const contractTracker = {};
+let timestampThreshold = 0; // The timestamp time in seconds based on the bucket block size and blockTime
+
+let contractsForChain = [];
+let isFirstBlock = true;
+let startTimestamp = 0;
+let isTrained = false;
+let isRunningJob = false;
+let localFindings = [];
+
+//Initialize the TSA for each contract in the list
+const initialize = async () => {
+  const { chainId } = await provider.getNetwork();
+  contractsForChain = getContractsByChainId(chainId);
+  blockTime = getBlocktimeByChainId(chainId);
+  const minBlockSize = getMinBucketBlockSizeByChainId(chainId);
+  if (bucketBlockSize < minBlockSize) {
+    console.warn(
+      `Min bucket block size for chainId: ${chainId} is ${minBlockSize}, setting it to that value`
+    );
+    bucketBlockSize = minBlockSize;
+  }
+  timestampThreshold = bucketBlockSize * blockTime;
+
+  const seasonality = 604_800 / timestampThreshold;
+  if (seasonality > 50) {
+    throw new Error(
+      "Seasonality cannot be greater than 50, please increase bucket block size"
+    );
+  }
```

Comment:
not sure I quite understand this one. Can you add a comment to clarify.

Reply:
If Seasonality is greater than 50, it takes an insanely high amount of time to train the model for predictions, so its optimal to keep the Seasonality max to 50, based on a weeks data, and depending on that we set the bucket block size and have a minimum value preset